NebulaPatcher/Patches/Dynamic/UIOptionWindow_Patch.cs

```diff
@@ -0,0 +1,279 @@
+        {
+            if (idx == multiplayerTabIndex)
+            {
+                tempMultiplayerOptions = new MultiplayerOptions();
+            }
+        }
+
+        [HarmonyPostfix]
+        [HarmonyPatch("TempOptionToUI")]
+        public static void TempOptionToUI_Postfix()
+        {
+            List<PropertyInfo> properties = AccessTools.GetDeclaredProperties(typeof(NebulaModel.MultiplayerOptions));
+            foreach (var prop in properties)
+            {
+                if (tempToUICallbacks.ContainsKey(prop.Name))
+                {
+                    tempToUICallbacks[prop.Name]();
+                }
+            }
+        }
+
+        private static void AddMultiplayerOptionsProperties(RectTransform container)
+        {
+            List<PropertyInfo> properties = AccessTools.GetDeclaredProperties(typeof(NebulaModel.MultiplayerOptions));
+            Vector2 anchorPosition = new Vector2(30, -20);
+
+            foreach (var prop in properties)
+            {
+                UIControlAttribute controlAttr = prop.GetCustomAttribute<UIControlAttribute>();
+                if (controlAttr != null)
+                {
+                    if (prop.PropertyType == typeof(bool))
+                    {
+                        CreateBooleanControl(controlAttr, prop, anchorPosition);
+                    }
+                    else if (prop.PropertyType == typeof(int) || prop.PropertyType == typeof(float))
+                    {
+                        CreateNumberControl(controlAttr, prop, anchorPosition);
+                    }
+                    else if (prop.PropertyType == typeof(string))
+                    {
+                        CreateStringControl(controlAttr, prop, anchorPosition);
+                    }
+                    else if (prop.PropertyType.IsEnum)
+                    {
+                        CreateEnumControl(controlAttr, prop, anchorPosition);
+                    }
+                    else
+                    {
+                        Log.Warn($"MultiplayerOption property \"${prop.Name}\" of type \"{prop.PropertyType}\" not supported.");
+                        continue;
+                    }
+
+                    anchorPosition = new Vector2(anchorPosition.x, anchorPosition.y - 40);
+                }
+            }
+        }
+
+        private static void CreateBooleanControl(UIControlAttribute control, PropertyInfo prop, Vector2 anchorPosition)
+        {
+            RectTransform element = Object.Instantiate(checkboxTemplate, multiplayerContent, false);
+            SetupUIElement(element, control, prop, anchorPosition);
+            UIToggle toggle = element.GetComponentInChildren<UIToggle>();
+            toggle.toggle.onValueChanged.RemoveAllListeners();
+            toggle.toggle.onValueChanged.AddListener((value) => { prop.SetValue(tempMultiplayerOptions, value, null); });
+
+            tempToUICallbacks[prop.Name] = () =>
```

Comment:
This might end up with a case where, due to GameObject.Destroy(), this dictionary will contain a null value. This dictionary should be cleared when the multiplayer content widgets are removed.